airflow/plugins_manager.py

```diff
@@ -41,33 +45,56 @@ def validate(cls):
 if plugins_folder not in sys.path:
     sys.path.append(plugins_folder)
 
-plugins = []
 
-# Crawl through the plugins folder to find AirflowPlugin derivatives
-for root, dirs, files in os.walk(plugins_folder, followlinks=True):
-    for f in files:
-        try:
-            filepath = os.path.join(root, f)
-            if not os.path.isfile(filepath):
-                continue
-            mod_name, file_ext = os.path.splitext(
-                os.path.split(filepath)[-1])
-            if file_ext != '.py':
-                continue
-            namespace = root.replace('/', '__') + '_' + mod_name
-            m = imp.load_source(namespace, filepath)
-            for obj in list(m.__dict__.values()):
-                if (
-                        inspect.isclass(obj) and
-                        issubclass(obj, AirflowPlugin) and
-                        obj is not AirflowPlugin):
-                    obj.validate()
-                    if obj not in plugins:
-                        plugins.append(obj)
+if 'plugins' not in globals():
```

Comment:
It guards the plugins from being loaded twice, which can cause problems depending on what the plugin code does. It seems natural to assume that plugins are only loaded once unless there is some explicit trigger like a file change.